v03_pipeline/lib/tasks/reference_data/updated_cached_reference_dataset_query.py

```diff
@@ -0,0 +1,127 @@
+import hail as hl
+import luigi
+
+from v03_pipeline.lib.model import (
+    CachedReferenceDatasetQuery,
+    Env,
+    ReferenceDatasetCollection,
+)
+from v03_pipeline.lib.paths import (
+    valid_cached_reference_dataset_query_path,
+    valid_reference_dataset_collection_path,
+)
+from v03_pipeline.lib.reference_data.compare_globals import (
+    Globals,
+    validate_globals_match,
+)
+from v03_pipeline.lib.reference_data.config import CONFIG
+from v03_pipeline.lib.reference_data.dataset_table_operations import (
+    get_ht_path,
+    import_ht_from_config_path,
+)
+from v03_pipeline.lib.tasks.base.base_write_task import BaseWriteTask
+from v03_pipeline.lib.tasks.files import GCSorLocalTarget, HailTableTask
+from v03_pipeline.lib.tasks.reference_data.updated_reference_dataset_collection import (
+    UpdatedReferenceDatasetCollectionTask,
+)
+
+
+class UpdatedCachedReferenceDatasetQuery(BaseWriteTask):
+    crdq = luigi.EnumParameter(enum=CachedReferenceDatasetQuery)
+
+    def complete(self) -> bool:
+        if not super().complete():
+            return False
+
+        crdq_globals = Globals.from_ht(
+            hl.read_table(self.output().path),
+            ReferenceDatasetCollection.COMBINED,
+            self.dataset_type,
+        )
+        dataset_config_globals = Globals.from_dataset_configs(
+            ReferenceDatasetCollection.COMBINED,
+            self.dataset_type,
+            self.reference_genome,
+        )
+        return validate_globals_match(
+            ReferenceDatasetCollection.COMBINED,
+            crdq_globals,
+            dataset_config_globals,
+            self.crdq.dataset,
+            validate_selects=False,
+        )
+
+    def output(self) -> luigi.Target:
+        return GCSorLocalTarget(
+            valid_cached_reference_dataset_query_path(
+                self.reference_genome,
+                self.dataset_type,
+                self.crdq,
+            ),
+        )
+
+    def requires(self) -> luigi.Task:
+        if Env.REFERENCE_DATA_AUTO_UPDATE and not self.crdq.query_raw_dataset:
+            return UpdatedReferenceDatasetCollectionTask(
+                self.reference_genome,
+                self.dataset_type,
+                self.sample_type,
+                ReferenceDatasetCollection.COMBINED,
+            )
+        if self.crdq.query_raw_dataset:
+            # NB: the config contents are not guaranteed to be hail tables/matrix tables.
```

Comment:
Your thoroughness actually fixed the bug here right so we don't need the comment any more right?

Reply:
yeah, the 2 raw datasets are a hail table and a matrix table